app/controllers/houseConsignment/index/departureMeansOfTransport/IdentificationNumberController.scala

```diff
@@ -0,0 +1,79 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.houseConsignment.index.departureMeansOfTransport
+
+import controllers.actions._
+import forms.VehicleIdentificationNumberFormProvider
+import models.{ArrivalId, Index, Mode}
+import navigation.Navigator
+import pages.houseConsignment.index.departureMeansOfTransport.VehicleIdentificationNumberPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewModels.departureTransportMeans.IdentificationNumberViewModel.IdentificationNumberViewModelProvider
+import views.html.houseConsignment.index.departureMeansOfTransport.IdentificationNumberView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class IdentificationNumberController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  actions: Actions,
+  formProvider: VehicleIdentificationNumberFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  navigator: Navigator,
+  view: IdentificationNumberView,
+  identificationNumberViewModelProvider: IdentificationNumberViewModelProvider
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, transportMeansIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId) {
+      implicit request =>
+        val viewModel = identificationNumberViewModelProvider.apply(mode)
+        val form      = formProvider(mode)
+
+        val preparedForm = request.userAnswers.get(VehicleIdentificationNumberPage(houseConsignmentIndex, transportMeansIndex)) match {
+          case None        => form
+          case Some(value) => form.fill(value)
+        }
+
+        Ok(view(preparedForm, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, transportMeansIndex, mode, viewModel))
+    }
+
+  def onSubmit(arrivalId: ArrivalId, houseConsignmentIndex: Index, transportMeansIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId).async {
+      implicit request =>
+        val viewModel = identificationNumberViewModelProvider.apply(mode)
+        val form      = formProvider(mode)
+        form
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              Future
+                .successful(BadRequest(view(formWithErrors, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, transportMeansIndex, mode, viewModel))),
+            value =>
+              for {
```

Comment:
maybe personal preference, but this looks quite big - maybe we could extract it to a method similar to [this](https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/main/app/controllers/houseConsignment/index/items/additionalReference/AdditionalReferenceNumberController.scala#L75).